aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -396,7 +396,12 @@ Tensor eye(int64_t n, int64_t m,
   // See [Note: hacky wrapper removal for TensorOptions]
   TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
 
-  auto tensor = at::empty({0}, options); // to be resized
+  Tensor tensor;
+  if (layout == kSparseCsr) {
```

Comment:
Why is conditioning on the `kSparseCsr` layout required, but not for `kSparse`?
Is it because at::empty fails for `{0}` shape for SparseCsr? If you try using
```
auto tensor = at::empty({0, 0}, options);
```
without  conditioning on the `layout` would it work?

Reply:
> Is it because at::empty fails for {0} shape for SparseCsr? If you try using

Yes, Ivan! That's right.

And yes, using `{0, 0}` works!! Thank you.